src/main/java/com/exactpro/th2/common/schema/factory/CommonFactory.java

```diff
@@ -112,80 +112,70 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CommonFactory.class.getName());
 
-    protected CommonFactory(
-            Class<? extends MessageRouter<MessageBatch>> messageRouterParsedBatchClass,
-            Class<? extends MessageRouter<RawMessageBatch>> messageRouterRawBatchClass,
-            Class<? extends MessageRouter<MessageGroupBatch>> messageRouterMessageGroupBatchClass,
-            Class<? extends MessageRouter<EventBatch>> eventBatchRouterClass,
-            Class<? extends MessageRouter<EventBatch>> notificationEventBatchRouterClass,
-            Class<? extends GrpcRouter> grpcRouterClass,
-            @Nullable Path custom,
-            @Nullable Path dictionariesDir,
-            @Nullable Path oldDictionariesDir,
-            Map<String, String> environmentVariables,
-            ConfigurationManager configurationManager
-    ) {
-        super(
+    /**
+     * @deprecated Please use {@link CommonFactory#CommonFactory(FactorySettings)}
+     */
+    @Deprecated(since = "4.0.0", forRemoval = true)
+    protected CommonFactory(Class<? extends MessageRouter<MessageBatch>> messageRouterParsedBatchClass,
+                            Class<? extends MessageRouter<RawMessageBatch>> messageRouterRawBatchClass,
+                            Class<? extends MessageRouter<MessageGroupBatch>> messageRouterMessageGroupBatchClass,
+                            Class<? extends MessageRouter<EventBatch>> eventBatchRouterClass,
+                            Class<? extends GrpcRouter> grpcRouterClass,
+                            @Nullable Path custom,
+                            @Nullable Path dictionariesDir,
+                            @Nullable Path oldDictionariesDir,
+                            Map<String, String> environmentVariables,
+                            ConfigurationManager configurationManager) {
+        this(new FactorySettings(
                 messageRouterParsedBatchClass,
                 messageRouterRawBatchClass,
                 messageRouterMessageGroupBatchClass,
                 eventBatchRouterClass,
-                notificationEventBatchRouterClass,
                 grpcRouterClass,
-                environmentVariables
-        );
-        this.custom = defaultPathIfNull(custom, CUSTOM_FILE_NAME);
-        this.dictionariesDir = defaultPathIfNull(dictionariesDir, DICTIONARY_DIR_NAME);
-        this.oldDictionariesDir = requireNonNullElse(oldDictionariesDir, CONFIG_DEFAULT_PATH);
-        this.configurationManager = configurationManager;
-        start();
+                NotificationEventBatchRouter.class,
+                environmentVariables,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
```

Comment:
Maybe add the default constructor without or only required params for the FactorySettings class

Reply:
I added three more secondary constructors to `FactorySettings`.
I can omit default parameters but not all to avoid call constructor from itself, so I added some additional null parameter to call main constructor.
And also it's neccesary to point parameter names in some places.
E.g.:
```
) : this(
    messageRouterParsedBatchClass,
    messageRouterRawBatchClass,
    messageRouterMessageGroupBatchClass,
    eventBatchRouterClass,
    grpcRouterClass,
    variables = variables,
    rabbitMQ = null
)
```

Reply:
As we discussed I changed `FactorySettings` to "builders" style